### PR TITLE
chore: Use departmental tsconfig

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?
Use [@guardian/tsconfig](https://github.com/guardian/csnx/tree/main/libs/%40guardian/tsconfig) to reduce the amount of configuration.

This config has highlighted some code paths not returning expected values. These are also fixed.

## Why?
- Less config
- An easier path to ESM... I think!
- Makes it easier to use packages that are now published as ESM only (e.g. [node-fetch](https://github.com/guardian/service-catalogue/pull/87#discussion_r1036853767))

### CODE REVIEW

File: packages/common/src/aws/kms.test.ts [hunk 1]
```diff
@@ -22,14 +22,4 @@ describe('decrypt', function () {
 
 		expect(actualOutput).toBe(expectedOutput);
 	});
-
-	it('returns undefined if the decrypt operation fails', async function () {
```
Comment (akash1810) [inline packages/common/src/aws/kms.test.ts:26]: This is testing an oddity of AWS's types, the code path should never happen, so removed the test.

File: packages/common/src/aws/kms.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ const kmsClient = new KMSClient({
 export const decrypt = async (
 	cipherText: string,
 	keyId: string,
-): Promise<string | undefined> => {
+): Promise<string> => {
```
Comment (akash1810) [inline packages/common/src/aws/kms.ts:11]: This type is easier for callers to handle; the `undefined` case should never happen (see added comment below).

File: packages/common/src/github/github.ts [hunk 3]
```diff
@@ -69,6 +69,8 @@ export const getOctokit = (config: GitHubConfig): Octokit => {
 					console.log(`Retrying after ${retryAfter} seconds!`);
 					return true;
 				}
+
+				return false;
```
Comment (akash1810) [inline packages/common/src/github/github.ts:73]: Ensure all code paths return a value (same for below).

File: packages/github-lens-api/src/controller/repos.ts [hunk 4]
```diff
@@ -7,22 +7,22 @@ export const getAllRepos = (
...
-		if (jsonResponse.length !== 0) {
-			res.status(200).json(jsonResponse);
-		} else {
-			return res.status(200).json({
-				searchString: searchString,
-				info: 'no results found in repos',
-			});
-		}
-	} else {
+	if (typeof req.query.name === 'undefined') {
```
Comment (akash1810) [inline packages/github-lens-api/src/controller/repos.ts:10]: Returning early is easier to read.

File: packages/services-api/src/app.ts [hunk 5]
```diff
@@ -67,13 +67,17 @@ export function buildApp(config: Config): Express {
 
 	router.get(
 		'/teams/:id',
-		asyncHandler(async (req: express.Request, res: express.Response) => {
-			const team = await galaxiesApi.getTeam(req.params.id);
-			const services = await servicesApi.forGithubOwner(team.primaryGithubTeam);
+		asyncHandler(
+			async (req: express.Request<{ id: string }>, res: express.Response) => {
```
Comment (akash1810) [inline packages/services-api/src/app.ts:71]: Type annotation added to `req`, to ensure `req.params.id` is type-safe.

File: packages/services-api/src/galaxies.ts [hunk 6]
```diff
@@ -89,12 +89,14 @@ export class S3GalaxiesApi implements GalaxiesApi {
 
 		const teamsMapping = mappingResp.payload;
 
-		// Add the (for now) missing fields.
-		return Object.keys(teams).map((teamId) => {
+		return Object.entries(teams).map(([id, team]) => {
```
Comment (akash1810) [inline packages/services-api/src/galaxies.ts:92]: `tsc` wasn't convinced `teams[teamId]` wasn't undefined when using `Object.keys`, it's happier with `Object.entries` 🤷🏽 .

File: packages/common/src/aws/kms.ts [hunk 7]
```diff
@@ -18,14 +18,18 @@ export const decrypt = async (
...
+			return message;
 		}
+
+		console.log('Ciphertext successfully decrypted');
+		return Buffer.from(Plaintext).toString();
 	} catch (e) {
-		console.error(`Decryption failed: ${(e as Error).message}`);
+		const message = `Decryption failed: ${(e as Error).message}`;
+		console.error(message);
+		throw new Error(message);
```
Comment (akash1810) [inline packages/common/src/aws/kms.ts:33]: If we get here, we should fail fast, so `throw` (returning `undefined` requires extra logic by the caller).

File: packages/common/src/aws/kms.ts [hunk 8]
```diff
@@ -18,14 +18,18 @@ export const decrypt = async (
 		const { Plaintext } = await kmsClient.send(decryptCommand);
 		console.log('Decrypt command sent successfully');
 
-		if (Plaintext) {
-			console.log('Ciphertext successfully decrypted');
-			const plaintextString = Buffer.from(Plaintext).toString();
-			return plaintextString;
-		} else {
-			console.error('Plaintext is missing from DecryptCommandOutput!');
+		if (!Plaintext) {
```
Comment (kenoir) [inline packages/common/src/aws/kms.ts:21]: 👌 nice one -  I wondered why I was forced to deal with this case by the type system!

Comment (kenoir) [inline packages/common/src/aws/kms.ts:21]: 👌 nice one -  I wondered why I was forced to deal with this case by the type system!

File: packages/cdk/bin/cdk.ts [hunk 9]
```diff
@@ -1,8 +1,8 @@
-import 'source-map-support/register';
-import { GuRootExperimental } from '@guardian/cdk/lib/experimental/constructs/root';
-import { CloudFormationLens } from '../lib/cloudformation-lens';
-import { GithubLens } from '../lib/github-lens';
-import { ServicesApi } from '../lib/services-api';
+import 'source-map-support/register.js';
+import { GuRootExperimental } from '@guardian/cdk/lib/experimental/constructs/root.js';
```
Comment (akash1810) [inline packages/cdk/bin/cdk.ts:2]: I _think_ once GuCDK is published as ESM, we won't need these explicit extensions.

Comment (kenoir) [reply packages/cdk/bin/cdk.ts:2]: Oh wowza, this is quite unpleasant.

**Review by kenoir — APPROVED** [review]

File: packages/cdk/bin/cdk.ts (hunk 9 above)
Comment (akash1810) [reply packages/cdk/bin/cdk.ts:2]: > Oh wowza, this is quite unpleasant.

Indeed!

**Comment by akash1810** [comment]
Deployed, and everything seem to be running still. Merging.

**Pull request merged**